crates/core_simd/src/libmf32.rs

```diff
@@ -0,0 +1,35 @@
+use crate::SimdF32;
+
+impl<const LANES: usize> SimdF32<LANES>
+where
+    Self: crate::LanesAtMost32,
+{
+    // This does not seem to be implemented, yet.
+    #[inline]
+    fn mul_add(self, mul: Self, add: Self) -> Self {
+        self * mul + add
+    }
+
+    /// Compute the sine of the angle in radians.
+    /// Result is accurate to 0.0000002.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use core_simd::SimdF32;
+    /// let x = SimdF32::<8>::splat(1.0);
+    /// assert!((x.sin() - SimdF32::<8>::splat((1.0_f32).sin())).abs().horizontal_max() < 0.0000002);
+    /// ```
+    #[inline]
+    pub fn sin(&self) -> Self {
```

Comment:
Hm. Would this implementation be shared for f64, same values but different types? If so, we could macro this for applying to both, though my math brain is feeling deeply skeptical that such is how it works at the moment. It's been A Minute since I did trigonometry.